File: src/invertBinaryTree.py
# Definition for a binary tree node.
from typing import Optional


class TreeNode:
     def __init__(self, val=0, left=None, right=None):
         self.val = val
         self.left = left
         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def helper(root):
            if root:
                root.left, root.right = helper(root.right), helper(root.left)
                return root
        return helper(root)

File: src/invertBinaryTree.java
class Solution {
    public TreeNode helper(TreeNode root) {
        if (root != null) {
            TreeNode temp = root.left;
            root.left = helper(root.right);
            root.right = helper(temp);
            return root;
        } else {
            return null;
        }
    }

    public TreeNode invertTree(TreeNode root) {
        return helper(root);
    }
}
